optuna/storages/_rdb/storage.py

```diff
@@ -1115,16 +1112,6 @@ def remove_session(self) -> None:
 
         self.scoped_session.remove()
 
-    def _finalize(self) -> None:
-
-        # This destructor calls remove_session to explicitly close the DB connection. We need this
-        # because DB connections created in SQLAlchemy are not automatically closed by reference
-        # counters, so it is not guaranteed that they are released by correct threads (for more
-        # information, please see the docstring of remove_session).
```

Comment:
However, this comment says that `_finalize` is needed 🤔 Even though, it is never called.